chips/lowrisc/src/i2c.rs

```diff
@@ -317,15 +318,16 @@ impl<'a> I2c<'_> {
                     break;
                 }
                 // Send the data
-                regs.fdata.write(FDATA::FBYTE.val(buf[i as usize] as u32));
+                regs.fdata
+                    .write(FDATA::FBYTE.val(*buf.get(i).unwrap_or(&0) as u32));
                 data_pushed = i;
             }
 
             // Check if we can send the last byte
             if regs.status.read(STATUS::FMTFULL) == 0 && data_pushed == (len - 1) {
                 // Send the last byte with the stop signal
                 regs.fdata
-                    .write(FDATA::FBYTE.val(buf[len as usize] as u32) + FDATA::STOP::SET);
```

Comment:
Not sure if it's better to panic in a case like this or to 'silently fail' and write 0?